compiler-plugin/src/main/java/io/ballerina/stdlib/websocket/plugin/AddWebSocketCodeTemplate.java

```diff
@@ -0,0 +1,134 @@
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocumentChange;
+import io.ballerina.tools.text.TextEdit;
+import io.ballerina.tools.text.TextRange;
+import org.wso2.ballerinalang.compiler.diagnostic.properties.NonCatProperty;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code action to add resource code snippet.
+ */
+public class AddWebSocketCodeTemplate implements CodeAction {
+
+    public static final String NODE_LOCATION = "node.location";
+
+    @Override
+    public List<String> supportedDiagnosticCodes() {
+        return List.of(PluginConstants.CompilationErrors.TEMPLATE_CODE_GENERATION_HINT.getErrorCode());
+    }
+
+    @Override
+    public Optional<CodeActionInfo> codeActionInfo(CodeActionContext codeActionContext) {
+        Diagnostic diagnostic = codeActionContext.diagnostic();
+        List<DiagnosticProperty<?>> properties = diagnostic.properties();
+        if (properties.isEmpty()) {
+            return Optional.empty();
+        }
+        DiagnosticProperty<?> diagnosticProperty = properties.get(0);
+        if (!(diagnosticProperty instanceof NonCatProperty) ||
+                !(diagnosticProperty.value() instanceof ServiceDeclarationNode)) {
+            return Optional.empty();
+        }
+
+        ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) diagnosticProperty.value();
+
+        CodeActionArgument locationArg = CodeActionArgument.from(NODE_LOCATION,
+                serviceDeclarationNode.location().lineRange());
+        return Optional.of(CodeActionInfo.from("Add resource", List.of(locationArg)));
+    }
+
+    @Override
+    public List<DocumentEdit> execute(CodeActionExecutionContext codeActionExecutionContext) {
+        LineRange lineRange = null;
+        for (CodeActionArgument argument : codeActionExecutionContext.arguments()) {
+            if (NODE_LOCATION.equals(argument.key())) {
+                lineRange = argument.valueAs(LineRange.class);
+            }
+        }
+
+        if (lineRange == null) {
+            return Collections.emptyList();
+        }
+
+        SyntaxTree syntaxTree = codeActionExecutionContext.currentDocument().syntaxTree();
+        NonTerminalNode node = findNode(syntaxTree, lineRange);
+        if (!(node instanceof ServiceDeclarationNode)) {
+            return Collections.emptyList();
+        }
+
+        ServiceDeclarationNode serviceDeclarationNode = (ServiceDeclarationNode) node;
+
+        List<TextEdit> textEdits = new ArrayList<>();
+
+        String insertResourceText = "\n\tresource function get .() returns websocket:Service|websocket:Error"
+                + "{\n\t\treturn new WsService(); \n\t}\n";
+        TextRange resourceTextRange = TextRange.from(serviceDeclarationNode.openBraceToken().textRange().endOffset(),
+                0);
+        String insertWsServiceText = "\n\nservice class WsService {\n" +
```

Comment:
Shall we extract both `insertResourceText` and `insertWsServiceText` into public constants ?

Reply:
Fixed with 51accb1